text/3488-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,374 @@
+what keywords are shown on the front page.
+
+- **Choice:** Categories can be [easier to pick from since the list is relatively
+short](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265619779),
+while with keywords the number of options is almost infinite. At the same
+time, having a short list can make it harder for some crates to categorize
+them because they don't fit in any of the existing categories.
+
+- **Bike-shedding:** As the original implementation acknowledged, having a fixed
+list of categories will turn into endless bike-shedding discussions on which
+categories should exist or not, while that time could be spent on more useful
+work instead.
+
+
+# Combinations
+
+**Keywords && Categories** describes the current implementation where
+crates.io supports both concepts at the same time.
+
+**Recommended Keywords** (aka. noteworthy keywords, blessed keywords, official
+keywords, …) are a system where users can create new keywords on their own, but
+the crates.io team can add descriptions and potentially other metadata to
+so-called "recommended" keywords. Such a system was also suggested in the
+original categories implementation pull request [here](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265593195)
+and [here](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265623340).
+
+**Comparison:**
+
+- **Confusion:** Having both categories and keywords can be confusing to users
+due to the [overlap](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265593195)
+of functionality in both concepts. Especially when searching for crates, it is
+not clear whether it's better to look for a matching category or a matching
+keyword. With just keywords it can potentially be confusing too if multiple
+matching keywords exist, though that can be countered to some degree by
+promoting one of them to a recommended keyword.
+
+- **Creation:** One big advantage of the "recommended keywords" concept is that
+keywords can be promoted to be "recommended" retroactively
+[based on usage data](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265597304).
+Categories need to be created by the crates.io team before they can be used by
+any crates, which leads to situations where such categories are not actually
+being used and [only contain a single crate](https://crates.io/categories/science::neuroscience).
+
+- **Duplicates:** With a list of recommended keywords the chance for duplicates
+can be reduced, but not eliminated completely. Categories have the advantage
+of not having that problem, but at the same time they are also restrict crate
+authors to only those categories that have been pre-selected by the crates.io
+team.
+
+- **Clarity:** Similar to reducing the number of duplicates, having a list of
+recommended keywords with associated descriptions can help improve clarity
+on which crates should be labeled with certain keywords. The same applies to
+categories as well.
```

Comment:
This sentence felt a bit weird to me -- here's my take on it, and we can probably meet in the middle somewhere.

```suggestion
- **Clarity:** In addition to having a list of recommending keywords to choose
from, providing descriptions alongside those keywords helps users decide which
crates should be labeled with those keywords too. The same applies to categories
as well.
```